### PR TITLE
vSphere folder handling: restrict by rootPath + Add E2E tests

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
This PR will make sure we filter VM folders by the configured root path & it adds a E2E test.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/provider/cloud/vsphere/provider_test.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
+// +build e2e
```
Comment (mrIncompetent) [inline api/pkg/provider/cloud/vsphere/provider_test.go]: @alvaroaleman Objections against this approach?
It felt like a very simple way to test the vCenter API without having to do the full cluster creation dance

Comment (mrIncompetent) [reply api/pkg/provider/cloud/vsphere/provider_test.go]: I'll add a presubmit if we agree that this is acceptable :)

Comment (alvaroaleman) [reply api/pkg/provider/cloud/vsphere/provider_test.go]: This is perfect, that we currently only have unit or full-blown e2e tests is subpar.

Would you mind using an `integration` build tag instead thought? While it arguably e2e tests the particular functionality, in the scope of the whole product this is not an e2e test and I would like to be able to differentiate this kind of tests from tests that actually create a cluster

Comment (alvaroaleman) [reply api/pkg/provider/cloud/vsphere/provider_test.go]: Oh and also just to be sure, would you mind extending the `test` maketarget to also build this?

Comment (mrIncompetent) [reply api/pkg/provider/cloud/vsphere/provider_test.go]: > Would you mind using an integration build tag instead thought?

:+1: 

> would you mind extending the test maketarget to also build this?

:+1:

Comment (mrIncompetent) [reply api/pkg/provider/cloud/vsphere/provider_test.go]: Done in https://github.com/kubermatic/kubermatic/pull/4068/commits/817ef96b949421519f6cd17dd5e12ea60b7584bb

**Comment by mrIncompetent** [comment]
/hold

Let me fixup the env names so they match what we use in prow

**Comment by mrIncompetent** [comment]
@alvaroaleman PTAL

infra PR incoming

**Comment by alvaroaleman** [comment]
/lgtm
/hold

Can we actually merge the presubmit with `optional: true` and verify on this PR that it works?

**Comment by mrIncompetent** [comment]
@alvaroaleman PTAL

**Pull request merged**